### PR TITLE
Tweak lon_lat example to have correct lon and lat lines

### PR DESCRIPTION
**Pull request by nabobalis**
Someone pointed out we had the wrong lon,lat lines. 

I also updated some text and renamed the file.

### CODE REVIEW

File: examples/plotting/lon_lat_lines.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 Drawing heliographic longitude and latitude lines
 =================================================
 
-How to draw your own (Stonyhurst) longitude and latitude lines.
+This example demonstrates how you can draw your own Stonyhurst
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:6]:
```suggestion
This example demonstrates how you can draw individual Stonyhurst
```

File: examples/plotting/lon_lat_lines.py [hunk 2]
```diff
@@ -16,38 +17,37 @@
 from sunpy.data.sample import AIA_171_IMAGE
 
 ###############################################################################
-# The purpose of this example is to demonstrate the coordinate transformations
-# that occur under the hood to show the heliographic grid lines of longitude
-# latitude. We first create the Map using the sample data.
+# We will demonstrate the coordinate transformations that occur under the
+# hood to show the heliographic grid lines of longitude latitude.
+# We first create a map using the sample data.
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:22]:
```suggestion
# We will first demonstrate the coordinate transformations that occur under the
# hood to show a heliographic grid line of longitude or latitude.
# Let's start with a map using the sample data.
```

File: examples/plotting/lon_lat_lines.py [hunk 3]
```diff
@@ -16,38 +17,37 @@
...
 
 aia = sunpy.map.Map(AIA_171_IMAGE)
 
 ###############################################################################
-# Let's first transform a single heliographic point coordinate.
+# Let's first specify a single heliographic point coordinate.
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:27]:
```suggestion
# We can define a heliographic coordinate for a single point.
```

File: examples/plotting/lon_lat_lines.py [hunk 4]
```diff
@@ -16,38 +17,37 @@
...
 
 stonyhurst_center = SkyCoord(12 * u.deg, 12 * u.deg,
                              frame=frames.HeliographicStonyhurst)
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:30]:
```suggestion
```suggestion
lon_value = 35 * u.deg
lat_value = 12 * u.deg
stonyhurst_frame = frames.HeliographicStonyhurst(obstime=aia.date)
point_in_stonyhurst = SkyCoord(lon=lon_value, lat=lat_value, frame=stonyhurst_frame)
```

File: examples/plotting/lon_lat_lines.py [hunk 5]
```diff
@@ -16,38 +17,37 @@
...
 
 ###############################################################################
-# Next we transform it into the coordinate frame of our map which is in
-# helioprojective coordinates.
+# Next we transform it into the coordinate frame of our map which is helioprojective.
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:33]:
```suggestion
# Next we transform it into the coordinate frame of our map, which is helioprojective.
```

File: examples/plotting/lon_lat_lines.py [hunk 6]
```diff
@@ -16,38 +17,37 @@
...
 
 hpc_stonyhurst_center = stonyhurst_center.transform_to(aia.coordinate_frame)
 print(hpc_stonyhurst_center)
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:36]:
```suggestion
point_in_hpc = point_in_stonyhurst.transform_to(aia.coordinate_frame)
print(point_in_hpc)
```

File: examples/plotting/lon_lat_lines.py [hunk 7]
```diff
@@ -16,38 +17,37 @@
...
 
 ###############################################################################
-# Now let's define two lines, one of longitude and one of of latitude.
+# Now let's define two lines, one of constant longitude and one of constant latitude.
 # We don't need to explicitly transform them to the coordinate frame of our
 # map because they will be transformed automatically when plotted.
 
 num_points = 100
 lat_value = 12 * u.deg
 lon_value = 35 * u.deg
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:45]:
```suggestion
```
Because I moved the definition earlier

File: examples/plotting/lon_lat_lines.py [hunk 8]
```diff
@@ -16,38 +17,37 @@
...
-lon0 = SkyCoord(np.linspace(-80, 80, num_points) * u.deg,
-                np.ones(num_points) * lon_value, frame=frames.HeliographicStonyhurst)
-lat0 = SkyCoord(np.ones(num_points) * lat_value,
-                np.linspace(-90, 90, num_points) * u.deg,
-                frame=frames.HeliographicStonyhurst)
+lon = SkyCoord(np.ones(num_points) * lon_value,
+               np.linspace(-90, 90, num_points) * u.deg,
+               frame=frames.HeliographicStonyhurst)
+lat = SkyCoord(np.linspace(-90, 90, num_points) * u.deg,
+               np.ones(num_points) * lat_value, frame=frames.HeliographicStonyhurst)
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:50]:
```suggestion
constant_lon = SkyCoord(lon_value, np.linspace(-90, 90, num_points) * u.deg,
                        frame=stonyhurst_frame)
constant_lat = SkyCoord(np.linspace(-90, 90, num_points) * u.deg, lat_value,
                        frame=stonyhurst_frame)
```
The single value will automatically be broadcast, so there's no need to explicitly create an array of constant value.

File: examples/plotting/lon_lat_lines.py [hunk 9]
```diff
@@ -16,38 +17,37 @@
...
-# Next we transform it into the coordinate frame of our map which is in
-# helioprojective coordinates.
+# Next we transform it into the coordinate frame of our map which is helioprojective.
 
 hpc_stonyhurst_center = stonyhurst_center.transform_to(aia.coordinate_frame)
 print(hpc_stonyhurst_center)
 
 ###############################################################################
-# Now let's define two lines, one of longitude and one of of latitude.
+# Now let's define two lines, one of constant longitude and one of constant latitude.
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:39]:
```suggestion
# Now let's define two lines, one of constant longitude and one of constant latitude,
# that pass through the previously defined coordinate point.
```

File: examples/plotting/lon_lat_lines.py [hunk 10]
```diff
@@ -16,38 +17,37 @@
...
-                np.linspace(-90, 90, num_points) * u.deg,
-                frame=frames.HeliographicStonyhurst)
+lon = SkyCoord(np.ones(num_points) * lon_value,
+               np.linspace(-90, 90, num_points) * u.deg,
+               frame=frames.HeliographicStonyhurst)
+lat = SkyCoord(np.linspace(-90, 90, num_points) * u.deg,
+               np.ones(num_points) * lat_value, frame=frames.HeliographicStonyhurst)
 
 ###############################################################################
 # Now let's plot the results. We'll overlay the autogenerated lon/lat
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:53]:
```suggestion
# Now let's plot the single coordinate point and the individual lines of
# constant longitude and latitude. We'll overlay the autogenerated lon/lat
```

File: examples/plotting/lon_lat_lines.py [hunk 11]
```diff
@@ -56,8 +56,8 @@
 fig = plt.figure()
 ax = fig.add_subplot(projection=aia)
 aia.plot(axes=ax, clip_interval=(1, 99.99)*u.percent)
-ax.plot_coord(lat0, color="C0")
-ax.plot_coord(lon0, color="C0")
+ax.plot_coord(lon, color="lightblue")
+ax.plot_coord(lat, color="tomato")
```
Comment (ayshih) [inline examples/plotting/lon_lat_lines.py:60]:
```suggestion
ax.plot_coord(lat, color="tomato")
ax.plot_coord(point_in_stonyhurst, marker="o")
```

**Review by ayshih — CHANGES_REQUESTED** [review]
A bunch of tweaks.

Also, I'm not inclined to rename the filename because that makes it much more cumbersome to see the same example at different versions of sunpy (e.g., by selecting different versions in our online docs).

**Comment by nabobalis** [comment]
> Also, I'm not inclined to rename the filename because that makes it much more cumbersome to see the same example at different versions of sunpy (e.g., by selecting different versions in our online docs).

Oh that is a good point. I will undo the name change.

File: examples/plotting/lat_lon_lines.py [hunk 12]
```diff
@@ -16,48 +17,49 @@
...
+# constant longitude and latitude. We'll overlay the autogenerated lon/lat
 # grid as well for comparison.
 
 fig = plt.figure()
 ax = fig.add_subplot(projection=aia)
 aia.plot(axes=ax, clip_interval=(1, 99.99)*u.percent)
-ax.plot_coord(lat0, color="C0")
-ax.plot_coord(lon0, color="C0")
+ax.plot_coord(lon, color="lightblue")
+ax.plot_coord(lat, color="tomato")
```
Comment (nabobalis) [inline examples/plotting/lat_lon_lines.py:61]:
```suggestion
ax.plot_coord(constant_lon, color="lightblue")
ax.plot_coord(constant_lat, color="tomato")
```

**Review by ayshih — APPROVED** [review]

**Pull request merged**